modules/core/src/v2/coins/stx.ts

```diff
@@ -110,10 +148,121 @@ export class Stx extends BaseCoin {
     }
   }
 
-  signTransaction(params: any): Bluebird<SignedTransaction> {
-    throw new Error('Method not implemented.');
+  /**
+   * Signs stacks transaction
+   * @param params
+   * @param callback
+   */
+  signTransaction(
+    params: StxSignTransactionOptions,
+    callback?: NodeCallback<SignedTransaction>
+  ): Bluebird<SignedTransaction> {
+    const self = this;
+
+    return co<SignedTransaction>(function*() {
```

Comment:
can we not use async/await here?

Reply:
No async await in implementation code yet please - we need to return a consistent promise type.